index.html

```diff
@@ -262,14 +263,19 @@
           <li>Remove a listener from an event.</li>
         </ul>
       </li>
+      <li>Retrieve a thing that is exposed by the local runtime</li>
       <li>Create and expose a local <a>Thing</a> based on a <a>Thing Description</a>.</li>
       <li>Programmatically create and expose a local <a>Thing</a>. This may include the following operations:
         <ul>
           <li>Add a property definition to the <a>Thing</a>.</li>
           <li>Add an event definition to the <a>Thing</a>.</li>
           <li>Add an action definition to the <a>Thing</a>.</li>
+          <li>Attach semanic information to an Action</li>
+          <li>Attach semanic information to a Property</li>
+          <li>Attach semanic information to an Event</li>
           <li>Start the exposed <a>Thing</a> in order to process external requests.</li>
-          <li>Handle an external request to retrieve the <a>Thing Description</a>.</li>
+          <li>Modify the handling of an external request to retrieve the <a>Thing Description</a>.</li>
+          <li>React on a property change</li>
```

Comment:
so far we have discussed it as it in the way a $watch works in Angular 1. That is to say, the callback*s* get called upon updates with newValue and oldValue as parameter.
This enables to have several callbacks reacting on a change, but does not enable modifying or denying the change (dumb setter).
Use case is e.g. I have an RGB light with 3 colors, and when someone changes the value of RED, I change the physical ledstate (and possibly do some side-effect).

I could think of a pattern like the "handler gets called with (newValue, oldValue) and returns a value that is assigned to the property" but that would limit it to one handler IMHO.